diesel/src/types/impls/primitives.rs

```diff
@@ -109,3 +109,27 @@ impl <'a, T: ?Sized, ST, DB> ::types::FromSqlRow<ST, DB> for Cow<'a, T> where
         FromSql::<ST, DB>::from_sql(row.take())
     }
 }
+
+use expression::bound::Bound;
+use expression::{AsExpression, Expression};
+impl <'a, T: ?Sized, ST> ::expression::AsExpression<ST> for Cow<'a, T> where
```

Comment:
empty line above? Also, `AsExpression` is in scope, no need to write absolute path